src/lib/sidenav/sidenav.ts

```diff
@@ -40,14 +39,49 @@ export class MdDuplicatedSidenavError extends MdError {
   host: {
     '(transitionend)': '_onTransitionEnd($event)',
     // must prevent the browser from aligning text based on value
-    '[attr.align]': 'null'
+    '[attr.align]': 'null',
+    '[class.md-sidenav-closed]': '_isClosed',
+    '[class.md-sidenav-closing]': '_isClosing',
+    '[class.md-sidenav-end]': '_isEnd',
+    '[class.md-sidenav-opened]': '_isOpened',
+    '[class.md-sidenav-opening]': '_isOpening',
+    '[class.md-sidenav-over]': '_modeOver',
+    '[class.md-sidenav-push]': '_modePush',
+    '[class.md-sidenav-side]': '_modeSide',
+    '[class.md-sidenav-invalid]': '!valid',
   },
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
 })
 export class MdSidenav implements AfterContentInit {
   /** Alignment of the sidenav (direction neutral); whether 'start' or 'end'. */
-  @Input() align: 'start' | 'end' = 'start';
+  private _align: 'start' | 'end' = 'start';
+
+  /** Whether this md-sidenav is part of a valid md-sidenav-layout configuration. */
+  get valid() {
+    return this._valid;
+  }
+  set valid(value) {
+    value = coerceBooleanProperty(value);
+    if (!value) {
+      this.close();
+    }
+    this._valid = value;
+  }
+  private _valid = true;
+
+  @Input()
+  get align() {
+    return this._align;
+  }
+  set align(value) {
+    // Make sure we have a valid value.
+    value = (value == 'end') ? 'end' : 'start';
+    if (value != this._align) {
+      this._align = value;
+      this.onAlignChanged.emit();
```

Comment:
Why not just directly call `validateDrawers` here instead of going through an observable?